// @andoshin11/storybook-vue3/lib/server/framework-preset-vue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.webpack = void 0;
const vue_loader_v16_1 = require("vue-loader-v16");
function webpack(config) {
    return {
        ...config,
        plugins: [...(config.plugins || []), new vue_loader_v16_1.VueLoaderPlugin()],
        module: {
            ...config.module,
            rules: [
                ...((config.module || { rules: [] }).rules || []),
                {
                    test: /\.vue$/,
                    loader: require.resolve('vue-loader-v16'),
                    options: {},
                },
                {
                    test: /\.tsx?$/,
                    use: [
                        {
                            loader: require.resolve('ts-loader'),
                            options: {
                                transpileOnly: true,
                                appendTsSuffixTo: [/\.vue$/],
                            },
                        },
                    ],
                },
            ],
        },
        resolve: {
            ...config.resolve,
            extensions: [...((config.resolve || {}).extensions || []), '.vue'],
            alias: {
                ...(config.resolve || {}).alias,
                vue$: require.resolve('vue/dist/vue.esm-bundler.js'),
            },
        },
    };
}
exports.webpack = webpack;
