// @andoshin11/storybook-vue3/lib/client/preview/index-legacy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.raw = exports.getStorybook = exports.forceReRender = exports.setAddon = exports.clearDecorators = exports.addParameters = exports.addDecorator = exports.configure = exports.storiesOf = exports.WRAPS = void 0;
/* eslint-disable prefer-destructuring */
const vue_1 = require("vue");
const { start } = require('@storybook/core/client');
require("./globals");
const render_1 = require("./render");
const util_1 = require("./util");
exports.WRAPS = 'STORYBOOK_WRAPS';
function prepare(rawStory, innerStory) {
    console.log('////////inside prepare ///////////////');
    let story;
    if (typeof rawStory === 'string') {
        story = { template: rawStory };
    }
    else if (rawStory != null) {
        story = rawStory;
    }
    else {
        return null;
    }
    // console.log(story)
    // @ts-ignore
    // console.log(typeof defineComponent(story))
    // @ts-ignore
    // eslint-disable-next-line no-underscore-dangle
    if (innerStory) {
        story.components = { story: innerStory };
        return story;
    }
    render_1.propsContainer.props = util_1.extractProps(story.props);
    return vue_1.defineComponent({
        render() {
            return vue_1.h(story, render_1.propsContainer.props);
        }
    });
    return story;
    const result = vue_1.defineComponent({
        // @ts-ignore // https://github.com/storybookjs/storybook/pull/7578#discussion_r307985279
        [exports.WRAPS]: story,
        // @ts-ignore // https://github.com/storybookjs/storybook/pull/7578#discussion_r307984824
        [VALUES]: { ...(innerStory ? innerStory.options[VALUES] : {}), ...util_1.extractProps(story) },
        functional: true,
        setup(props) {
            return vue_1.h(story);
        },
    });
    return vue_1.defineComponent({
        // // @ts-ignore // https://github.com/storybookjs/storybook/pull/7578#discussion_r307985279
        // [WRAPS]: story,
        // // @ts-ignore // https://github.com/storybookjs/storybook/pull/7578#discussion_r307984824
        // [VALUES]: { ...(innerStory ? innerStory.options[VALUES] : {}), ...extractProps(story) },
        // // components: { MyButton },
        // template: '<button @click="action">Hello Button</button>',
        setup() {
            console.log('inside setup');
        },
    });
    // console.log('created instance')
    // console.log(result)
    return result;
}
const defaultContext = {
    id: 'unspecified',
    name: 'unspecified',
    kind: 'unspecified',
    parameters: {},
    args: {},
    argTypes: {},
    globals: {},
};
function decorateStory(storyFn, 
// @ts-ignore
decorators
// @ts-ignore
) {
    const result = decorators.reduce(
    // @ts-ignore
    (decorated, decorator) => (context = defaultContext) => {
        let story;
        const decoratedStory = decorator(({ parameters, ...innerContext } = {}) => {
            story = decorated({ ...context, ...innerContext });
            return story;
        }, context);
        if (!story) {
            story = decorated(context);
        }
        if (decoratedStory === story) {
            return story;
        }
        return prepare(decoratedStory, story);
    }, (context) => prepare(storyFn(context)));
    return result;
}
const framework = 'vue';
const api = start(render_1.default, { decorateStory });
exports.storiesOf = (kind, m) => {
    return api.clientApi.storiesOf(kind, m).addParameters({
        framework,
    });
};
exports.configure = (...args) => api.configure(framework, ...args);
exports.addDecorator = api.clientApi.addDecorator;
exports.addParameters = api.clientApi.addParameters;
exports.clearDecorators = api.clientApi.clearDecorators;
exports.setAddon = api.clientApi.setAddon;
exports.forceReRender = api.forceReRender;
exports.getStorybook = api.clientApi.getStorybook;
exports.raw = api.clientApi.raw;
